ingest/indexer/service/blockprocessor/full_indexer_block_process_strategy.go

```diff
@@ -0,0 +1,113 @@
+package blockprocessor
+
+import (
+	"sync"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	commondomain "github.com/osmosis-labs/osmosis/v25/ingest/common/domain"
+	"github.com/osmosis-labs/osmosis/v25/ingest/indexer/domain"
+)
+
+type fullIndexerBlockProcessStrategy struct {
+	client            domain.Publisher
+	keepers           domain.Keepers
+	poolExtractor     commondomain.PoolExtractor
+	poolPairPublisher domain.PairPublisher
+}
```

Comment:
Note: unit tests for this file are tracked here: https://linear.app/osmosis/issue/IND-71/unit-test-ingestindexerserviceblockprocessorfull-indexer-block-process